lib/bolt/inventory/group2.rb

```diff
@@ -115,6 +77,79 @@ def target_data(target_name)
         end
       end
 
+      def add_target(target)
+        # TODO: Do we want to accept strings from lookup_targets plugins? How should
+        # they be handled?
+        unless target.is_a?(Hash)
+          raise ValidationError.new("Node entry must be a String or Hash, not #{target.class}", @name)
+        end
+
+        target['name'] ||= target['uri']
+
+        if target['name'].nil? || target['name'].empty?
+          raise ValidationError.new("No name or uri for target: #{target}", @name)
+        end
+
+        if @targets.include?(target['name'])
+          @logger.warn("Ignoring duplicate target in #{@name}: #{target}")
+          return
+        end
+
+        raise ValidationError.new("Node #{target} does not have a name", @name) unless target['name']
+        @targets[target['name']] = target
+
+        unless (unexpected_keys = target.keys - NODE_KEYS).empty?
+          msg = "Found unexpected key(s) #{unexpected_keys.join(', ')} in target #{target['name']}"
+          @logger.warn(msg)
+        end
+        config_keys = target['config']&.keys || []
```

Comment:
I'm don't think it was introduced here, but perhaps we should validate that `target['config']` is either a `Hash` or `nil`. I accidentally did `config: local` and got a stack trace. 
```
NoMethodError: undefined method `keys' for "local":String
  /home/cas/working_dir/bolt/lib/bolt/inventory/group2.rb:106:in `add_target'
  /home/cas/working_dir/bolt/lib/bolt/inventory/group2.rb:59:in `block in initialize'
```

Reply:
This likely needs to be addressed in inventory v1 as well. Do not want to block this work on that. Filed https://tickets.puppetlabs.com/browse/BOLT-1274